src/pages/createpost/CreatePost.tsx

```diff
@@ -0,0 +1,111 @@
+import { IcRoundAdd } from "assets/Icons";
+import { FC, useRef, useState } from "react";
+import { useDispatch } from "react-redux";
+import { useNavigate } from "react-router-dom";
+import { addNewPost } from "reducers/postsSlice";
+import { useAxios } from "utils";
+import TextareaAutosize from "react-textarea-autosize";
+import { Grid as Loader } from "react-loader-spinner";
+import "./createpost.css";
+
+export const CreatePost: FC = () => {
+  const [img, setImg] = useState<string | undefined>();
+  const [selectedImage, setSelectedImage] = useState<Blob | undefined>();
+  const [imgLoader, setImgLoader] = useState(false);
+  const [caption, setCaption] = useState("");
+  const { loading, operation } = useAxios();
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const fileInput = useRef<HTMLInputElement>(null);
+
+  const handlePostPhoto = () => {
+    fileInput.current?.click();
+  };
+
+  const handleImageSelected = async () => {
+    if (fileInput.current?.files) {
+      const postPic = fileInput.current.files[0];
+      setImg(URL.createObjectURL(postPic));
+      setSelectedImage(postPic);
+    }
+  };
+
+  const handleUploadPost = () => {
+    if (selectedImage) {
+      const data = new FormData();
+      data.append("file", selectedImage);
+      data.append(
+        "upload_preset",
+        process.env.REACT_APP_CLOUDINARY_UPLOAD_PRESET ?? ""
+      );
+
+      setImgLoader(true);
+      fetch(process.env.REACT_APP_CLOUDINARY_API_URL ?? "", {
+        method: "post",
+        body: data,
+      })
+        .then((res) => res.json())
+        .then(async (data) => {
+          setImgLoader(false);
+          const response = await operation({
+            method: "post",
+            url: "/createpost",
+            data: { photo: data?.secure_url, caption },
+          });
+          if (response) {
+            dispatch(addNewPost(response.post));
+            navigate("/");
+          }
+        })
+        .catch((err) => {
+          setImgLoader(false);
+          console.log("err", err);
+        });
+    }
+  };
```

Comment:
Just curious why you used fetch rather than axios any specific reason

Reply:
My axios has auth headers set and cloudinary does not allow that. That's why i used fetch here.